examples/adder/src/lib.rs

```diff
@@ -75,13 +75,18 @@ mod tests {
 
         assert_eq!(add_function.name, "add".to_string());
         assert_eq!(add_function.doc, Some(" Adds two pairs point-wise.".to_string()));
-        assert!(add_function.is_view);
-        assert!(!add_function.is_init);
-        assert!(!add_function.is_payable);
-        assert!(!add_function.is_private);
-        assert_eq!(add_function.params.len(), 2);
-        assert_eq!(add_function.params[0].name, "a".to_string());
-        assert_eq!(add_function.params[1].name, "b".to_string());
+        assert_eq!(add_function.kind, AbiFunctionKind::View);
+        assert_eq!(add_function.modifiers, vec![]);
+        match &add_function.params {
+            AbiParameters::Json { args } => {
+                assert_eq!(args.len(), 2);
+                assert_eq!(args[0].name, "a".to_string());
+                assert_eq!(args[1].name, "b".to_string());
+            }
+            AbiParameters::Borsh { .. } => {
+                assert!(false);
```

Comment:
_nit_: [`unreachable!()`](https://doc.rust-lang.org/std/macro.unreachable.html)?

Reply:
Well, it is not actually unreachable if the thing we are trying to test happens to have borsh serialization though

Reply:
Hm, but isn't that the point of the test? Asserting that it is JSON serialized and not borsh?

Reply:
Yeah, but `unreacheable!` to me is something you know (statically) will never be reached, but can't convince compiler to be the case. This test, on the other hand, depends on some external input (a wasm file) and tests its properties.

This is arguing very subtle semantics and honestly it feels like `AbiParameters` should just have `json()?` and `borsh()?` methods instead of this match/fail pattern.

Reply:
That's fair, wasn't a strong opinion. I'm not sure if we're at the point where we need those helper methods, so we can proceed as-is. 👍🏽

Reply:
In that case, still seems like panic!() might be more readable?